odoo/addons/test_new_api/tests/test_properties.py

```diff
@@ -0,0 +1,1217 @@
+
+    def test_properties_field_delete(self):
+        """Test to delete a property using the flag "definition_deleted"."""
+        self.message_1.custom_properties = [{
+            'id': 'discussion_color_code',
+            'string': 'Test color code',
+            'type': 'char',
+            'default': 'blue',
+            'value': 'purple',
+        }, {
+            'id': 'moderator_partner_id',
+            'string': 'Partner',
+            'type': 'many2one',
+            'model': 'res.partner',
+            'value': [self.partner.id, 'Bob'],
+            'definition_deleted': True,
+        }]
+
+        sql_definition = self._get_sql_definition(self.discussion_1)
+        self.assertEqual(
+            sql_definition, [{
+                'id': 'discussion_color_code',
+                'type': 'char',
+                'string': 'Test color code',
+                'default': 'blue',
+            }])
+
+        self.assertEqual(len(self.message_1.custom_properties), 1)
+        self.assertEqual(self.message_1.custom_properties[0]['value'], 'purple')
+
+    def test_properties_field_create_batch(self):
+        with self.assertQueryCount(5):
+            messages = self.env['test_new_api.message'].create([{
+                'name': 'Test Message',
+                'discussion': False,
+                'author': self.user.id,
+            }, {
+                'name': 'Test Message',
+                'discussion': False,
+                'author': self.user.id,
+            }])
+            self._flush_and_invalidate()
+
+        with self.assertQueryCount(9):
+            messages = self.env['test_new_api.message'].create([{
+                'name': 'Test Message',
+                'discussion': self.discussion_1.id,
+                'author': self.user.id,
+                'custom_properties': [{
+                    'id': 'discussion_color_code',
+                    'string': 'New Label',
+                    'type': 'char',
+                    'default': 'blue',
+                    'value': 'purple',
+                }, {
+                    'id': 'moderator_partner_id',
+                    'string': 'Partner',
+                    'type': 'many2one',
+                    'model': 'res.partner',
+                    'value': [self.partner.id, 'Bob'],
+                    'definition_changed': True,
+                }],
+            }, {
+                'name': 'Test Message',
+                'discussion': self.discussion_2.id,
+                'author': self.user.id,
+                'custom_properties': [{
+                    'id': 'state',
+                    'type': 'selection',
+                    'string': 'Status',
+                    'selection': [
+                        ('draft', 'Draft'),
+                        ('progress', 'In Progress'),
+                        ('done', 'Done'),
+                    ],
+                    'default': 'draft',
+                    'definition_changed': True,
```

Comment:
Simply guessing what it does, it looks like a good way to screw up your data without noticing IMO.